src/main/java/jenkins/plugin/randomjobbuilder/GeneratorController.java

```diff
@@ -0,0 +1,365 @@
+@Extension
+public class GeneratorController extends RunListener<Run> {
+    private boolean autostart = false;
+
+    // FIXME the accounting for tasks is broken, sometimes it overcounts, sometimes it undercounts.
+    ConcurrentHashMap<String, LoadGenerator> registeredGenerators = new ConcurrentHashMap<String, LoadGenerator>();
+
+    /** Map {@link LoadGenerator#generatorId} to that LoadGenerator's queued item count */
+    ConcurrentHashMap<String, AtomicInteger> queueTaskCount = new ConcurrentHashMap<>();
+
+    /** Map {@link LoadGenerator#generatorId} to that LoadGenerator's {@link Run}s,
+     *   needed in order to cancel runs in progress */
+    ConcurrentHashMap<String, Collection<Run>> generatorToRuns = new ConcurrentHashMap<>();
+
+    /** Register the generator in {@link #registeredGenerators} or
+     *   replace existing generator with the same generator ID
+     *   Eventually we'll just use {@link ReconfigurableDescribable} for direct updates of generator lists.
+     *   @param generator Generator to register/update
+     */
+    public void registerOrUpdateGenerator(@Nonnull LoadGenerator generator) {
+        LoadGenerator previous = registeredGenerators.put(generator.getGeneratorId(), generator);
+        synchronized (generator) {
+            if (previous != null) {  // Copy some transitory state in, but honestly this is a hack
+                generator.copyStateFrom(previous);
+            }
+        }
+    }
+
+    /** Returns a snapshot of current registered load generators */
+    public List<LoadGenerator> getRegisteredGenerators() {
+        return new ArrayList<LoadGenerator>(registeredGenerators.values());
+    }
+
+    /**
+     * Unregister the generator and stop all jobs and tasks from it
+     * @param generator Generator to unregister/remove
+     */
+    public void unregisterAndStopGenerator(@Nonnull LoadGenerator generator) {
+        synchronized (generator) {
+            generator.stop();
+            this.stopAbruptly(generator);
+            registeredGenerators.remove(generator.getGeneratorId());
+        }
+    }
+
+    /** Find generator by its unique ID or return null if not registered
+     * @param generatorId ID of registered generator
+     */
+    @CheckForNull
+    public LoadGenerator getRegisteredGeneratorbyId(@Nonnull String generatorId) {
+        return registeredGenerators.get(generatorId);
+    }
+
+    /** Find generator by its unique ID or return null if not registered
+     * @param shortName ID of registered generator
+     */
+    @CheckForNull
+    public LoadGenerator getRegisteredGeneratorbyShortName(@Nonnull String shortName) {
+        for (LoadGenerator lg : registeredGenerators.values()) {
+            if (lg.getShortName().equals(shortName)) {
+                return lg;
+            }
+        }
+        return null;
+    }
+
+    /** Ensure that the registered generators match input set, registering any new ones and unregistering ones not in input,
```

Comment:
To be used with the UI heteroList in the resources/LoadGeneration/global.jelly so we don't lose internal load generator state info